src/pages/workspaces/workspace/analysis/Analyses.ts

```diff
@@ -230,163 +251,143 @@ const AnalysisCard = ({
   ])
 }
 
-const Analyses = _.flow(
-  forwardRefWithName('Analyses'),
-  requesterPaysWrapper({
-    onDismiss: () => Nav.history.goBack()
-  }),
-  wrapWorkspace({
-    breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
-    title: 'Analyses',
-    activeTab: 'analyses'
-  }),
-  withViewToggle('analysesTab')
-)(({
-  name: workspaceName, namespace, workspace, workspace: { accessLevel, canShare, workspace: { cloudPlatform, workspaceId, googleProject, bucketName } },
+const activeFileTransferMessage: React.ReactElement<'div'> = div({
+  style: _.merge(
+    Style.elements.card.container,
+    { backgroundColor: colors.warning(0.15), flexDirection: 'none', justifyContent: 'start', alignItems: 'center' })
+}, [
+  icon('warning-standard', { size: 19, style: { color: colors.warning(), flex: 'none', marginRight: '1rem' } }),
+  'Copying 1 or more interactive analysis files from another workspace.',
+  span({ style: { fontWeight: 'bold', marginLeft: '0.5ch' } }, ['This may take a few minutes.'])
+])
+
+export type DisplayAnalysisFile = AnalysisFile & {
+  application: ToolLabel
+}
+
+export const decorateAnalysisFiles = (rawAnalyses: AnalysisFile[]): DisplayAnalysisFile[] => {
+  const notebooks: AnalysisFile[] = _.filter(({ name }) => _.includes(getExtension(name), runtimeTools.Jupyter.ext), rawAnalyses)
+  const rAnalyses: AnalysisFile[] = _.filter(({ name }) => _.includes(getExtension(name), runtimeTools.RStudio.ext), rawAnalyses)
+
+  //we map the `toolLabel` corresponding header label, which simplifies the table sorting code
+  const enhancedNotebooks: DisplayAnalysisFile[] = _.map(file => ({ application: tools.Jupyter.label, ...file }), notebooks)
+  const enhancedRmd: DisplayAnalysisFile[] = _.map(file => ({ application: tools.RStudio.label, ...file }), rAnalyses)
+
+  const tempAnalyses: DisplayAnalysisFile[] = _.concat(enhancedNotebooks, enhancedRmd)
+  return _.reverse(_.sortBy(tableFields.lastModified, tempAnalyses))
+}
+
+export const getUniqueFileName = (originalName: string, existingFileNames: FileName[]): FileName => {
+  const name: FileName = originalName as FileName
+  let resolvedName: FileName = name
+  let c = 0
+  while (_.includes(resolvedName, existingFileNames)) {
+    resolvedName = `${getDisplayName(name)}_${++c}.${getExtension(name)}` as FileName
+  }
+  return resolvedName
+}
+
+export interface AnalysesData {
+  apps: App[]
+  refreshApps: () => Promise<void>
+  runtimes: Runtime[]
+  refreshRuntimes: () => Promise<void>
+  appDataDisks: PersistentDisk[]
+  persistentDisks: PersistentDisk[]
+}
+
+export interface AnalysesProps {
+  workspace: WorkspaceWrapper
+  analysesData: AnalysesData
+  onRequesterPaysError: () => void
+  storageDetails: StorageDetails
+}
+
+export interface SortOrderInfo {
+  field: string
+  direction: string
```

Comment:
Is there a more specific type available for this? Something like `'asc' | 'desc'`?